### ISSUE TITLE
Assignment type annotation with `Callable` is inconsistent with mypy

### ISSUE THREAD

**Issue by Drino** [microsoft/pyright#3806]
**Describe the bug**
`identity: Callable[[T], T] = lambda x: x` works as expected in mypy and fails in pyright.

**To Reproduce**
```python
from typing import TypeVar, Dict, Callable
from typing_extensions import reveal_type

T = TypeVar('T')

identity: Callable[[T], T] = lambda x: x

reveal_type(identity)
reveal_type(identity(3))
```

With mypy 
> main.py:8: note: Revealed type is "def [T] (T`-1) -> T`-1"
> main.py:9: note: Revealed type is "builtins.int"

With pyright:
> Type variable "T" has no meaning in this context
> Type of "identity" is "(x: T) -> T"
> Type of "identity(3)" is "T"

**Expected behavior**
I'd expect no warnings, as T has meaning in this context and correct type inference:
> Type of "identity(3)" is "int"

**VS Code extension or command-line**
Both. In VS Code I have Pylance v2022.8.10

**Comment by erictraut** [comment]
I consider mypy's behavior incorrect in this case.

For valid use, a type variable must be scoped to a function declaration (i.e. a `def` statement), a class declaration, or a type alias. You are using `T` here in a context where there is no defined scope.

Pyright correctly flags the use of `T` as having no meaning in this context.

Many Python developers are confused by the scoping rules for type variables. We're trying to clean this up with PEP 695, which will make the scope clear.
